index.html

```diff
@@ -893,6 +1044,47 @@ <h4 class="modal-title">Help</h4>
                         scene.add(piece);
                     }
                 },
+                updateboard: function () {
+                    // reload textures.
+                    var white_texture = THREE.ImageUtils.loadTexture(white_square_tex_name);
+                    var black_texture = THREE.ImageUtils.loadTexture(black_square_tex_name);
+                    
+                    var white_material = new THREE.MeshBasicMaterial({map: white_texture});
+                    var black_material = new THREE.MeshBasicMaterial({map: black_texture});
+                    
+                    // the first size^2 board_pieces are the squares.
+                    var fields = this.size * this.size;
+                    for (i = 0; i < fields; ++i) {
+                        if (this.board_objects[i].isboard===true) {
+                            this.board_objects[i].material = (i % 2) ? white_material : black_material;
+                        }
+                    }
+                },
+                updatepieces: function () {
+                    var geometry = new THREE.BoxGeometry(piece_size, piece_height, piece_size);
+                    var old_size = this.piece_objects[0].geometry.parameters.width;
+                    
+                    // for all pieces...
+                    for (i = 0; i < this.piece_objects.length; ++i) {
+                        if (this.piece_objects[i].iscapstone) {
+                        // <updates on the capstones will be made here>
+                        } else {
+                            // if standing, reset and reapply orientation.
+                            if (this.piece_objects[i].isstanding) {
+                                this.piece_objects[i].rotation.set(0, 0, 0);
+                                this.piece_objects[i].updateMatrix();
+                                this.piece_objects[i].position.y -= old_size / 2 - piece_height / 2;
+                                this.piece_objects[i].isstanding = false;
+                                this.standup(this.piece_objects[i]);
+                            }
+                            
+                            // reapply geometry.
+                            this.piece_objects[i].geometry = geometry;
+                            this.piece_objects[i].geometry = geometry;
+                            this.piece_objects[i].updateMatrix();
+                        }
+                    }
+                },
```

Comment:
update functions for board and pieces. They are currently separate, though called together. Should remain like this for clarity.